### PR TITLE
Create a staging slot to deploy to for zero-downtime

### PR DESCRIPTION
**Pull request by erbridge**
The runtime stack is still declared for the "production" slot so an incomplete deployment followed by a service restart doesn't take the service down. If not specified, the fact that we do a `Complete` deployment would mean that the configuration would be cleared when deploying, making the app fragile to a restart.

The app service doesn't change which version it's running unless it's restarted.

This situation might be improved by setting the main slot to use the currently live version explicitly, so configuration would match what's already live, though this wouldn't stop environment variable changes getting deployed. We could do this by querying the Azure API. I don't believe this is a good idea, however. It's better to deploy a complete configuration of the newest version accidentally than to deploy mixed configuration.

**BEFORE MERGE: This requires changing _both_ Azure DevOps deployment to (re)start the staging slot (possibly unnecessary, but it feels more reliable to be explicit), then swap staging to production.** It looks like you can type into the slot name, so this _can_ be done before merge, even though the slot doesn't exist. I **highly** recommend we set up production at the same time as we set up development, because release configuration gets attached to releases, so you can't easily change it after the development release occurs.

Additional reading:
- https://blogs.msdn.microsoft.com/mvpawardprogram/2017/05/16/deploy-app-azure-app-service/
- https://ruslany.net/2015/09/how-to-warm-up-azure-web-app-during-deployment-slots-swap/

### CODE REVIEW

**Comment by erbridge** [comment]
[Use slots to warm up app service changes without downtime](https://trello.com/c/TGVFnHqB/417-use-slots-to-warm-up-app-service-changes-without-downtime)

File: bin/swap-slots [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+RESOURCE_GROUP=$1
+APP_NAME="$RESOURCE_GROUP-as"
+
+az webapp deployment slot swap --name "$APP_NAME" --slot staging --resource-group "$RESOURCE_GROUP"
```
Comment (erbridge) [inline bin/swap-slots]: Should this also restart the app service, then?

Comment (pezholio) [reply bin/swap-slots]: Yeah, probably makes sense to do that too

File: bin/swap-slots [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    exit 1
+    ;;
+esac
+
+while [ "$2" ]; do
+  case $2 in
+    "--tmp-app")
+      RESOURCE_GROUP="$RESOURCE_GROUP-tmp"
+      ;;
+  esac
```
Comment (erbridge) [inline bin/swap-slots]: This should have a `*)` case to warn about unexpected arguments. See `azure-deploy`.

**Pull request merged**